metadata/docs/configurations.md

```diff
@@ -34,6 +34,31 @@ def get_user_details(user_id):
 USER_DETAIL_METHOD = get_user_details
 ```
 
+#### Flask Hooks
+
+Flask offers several hooks that can be used to execute before and after a request as well as before the first request and during app teardown. Amundsen allows you to provide an arbitrary function that can be called by these Flask hooks by providing the funtion as a configuration in your app.
+
+Note - some Flask hooks have required function signatures and the funciton you supply should match this requirement. For more information on each function signature visit Flask documentation for [before_first_request](https://flask.palletsprojects.com/en/2.0.x/api/#flask.Flask.before_first_request), [before_request](https://flask.palletsprojects.com/en/2.0.x/api/#flask.Flask.before_request), [after_request](https://flask.palletsprojects.com/en/2.0.x/api/#flask.Blueprint.after_request) and [teardown_request](https://flask.palletsprojects.com/en/2.0.x/api/#flask.Flask.teardown_request).
+
+```python
+from flask import request
+
+def verify_headers():
+    my_auth = request.headers.get('Authorization')
+    print(my_auth)
+
+
+def streap_resp(resp):
+    resp_data = resp.json
+    stream_data_to_kafka(resp_data)
```

Comment:
we have the event action logging framework used at Lyft since 3 years ago (https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/log/action_log.py) . Any reason we need another hook in the metadata layer?

Reply:
The specific implementation for this is to be able to use neo4j as a "cache" of sorts for user information. First, call the GET User API and, if the response is a 404 then go and reach out to the OIDC provider to get the user information. In addition, we check other attributes (e.g. has it been more than 30 days since we got the user information from the OIDC provider) and determine if we need to get current information from the OIDC provider again.

The examples provided were just intended to be ways that this could be used. While that particular example does overlap heavily with the action logger, this approach allows for more interaction with the Flask objects pre and post execution.

Reply:
I could be wrong, but I think this use case is a bit specific (caching user information other than what we have in user model)  which I am not sure how open source is used in general. Could this feature be done through private fork?

Reply:
e.g lots of company (including Lyft) has private proxy on top of open source proxy to add some custom changes.

Reply:
Yea, I think that's fair. I'll close this out. Should someone else happen to broach this how-to in the future we can continue the topic there.

Reply:
thanks